### PR TITLE
Wizard hat  bb   compile error changes

### PR DESCRIPTION
**Pull request by Wirednuke**
Working with Chilly to make changes based on compile errors. 

@jkachhad Still looking for some input/review.

### CODE REVIEW

**Comment by Wirednuke** [comment]
Compile errors for reference:

A segment map could not be compiled due to the following errors:
Compile Segment Exception (5 of 6 errors)
[1] Cache - 'Cache.WizardHat.getMagicskill(MobileEntity)': virtual or abstract members cannot be private -         private virtual double getMagicskill(MobileEntity entity)
[2] Cache - 'Cache.WizardHat.getMagicProtection(MobileEntity)': no suitable method found to override -         public override int getMagicProtection(MobileEntity entity)
[3] Cache - The name 'skillLevelrnd' does not exist in the current context -             return skillLevelrnd;
[4] Cache - The name 'MagicProtection' does not exist in the current context -                 MagicProtection = 2 * getMagicskill(entity);
[5] Cache - The name 'MagicProtection' does not exist in the current context -             return MagicProtection;

**Review by Chilly021 — COMMENTED** [review]
public override int MagicProtection => getMagicskill(Parent);

File: Segments/Blackburrow.mapproj [hunk 1]
```diff
@@ -142,6 +142,113 @@
...
+        
+        /// <inheritdoc />
+        public override uint BasePrice => 12500;
+
+        /// <inheritdoc />
+        public override bool CanBind => true;
+        
+        public override int ProtectionFromStun => 10;
+
+        public override int MagicProtection => getMagicskill();
```
[the remark this answers is not in the document]
Comment (Wirednuke) [reply Segments/Blackburrow.mapproj:161]: Yep, I see that now. I misunderstood Lux's direction. 

Inside my getMagicProtection method can I use Parent the same way I would any other class object? i.E. Parent.GetMagicSkillLvl(Skill.Magic); etc etc.

File: Segments/Blackburrow.mapproj [hunk 2]
```diff
@@ -142,6 +142,113 @@
...
+            
+            var skillLevel = entity.GetSkillLevel(Skill.Magic);
+            int skilllevelrnd = Math.Round(skillLevel);
+
+            return skillLevelrnd;
+        }
+
+        public int getMagicProtection(MobileEntity entity)
+        {
+
```
Comment (Chilly021) [inline Segments/Blackburrow.mapproj:175]: if (entity == null)
{
  return 0;
 }

**Review by Chilly021 — COMMENTED** [review]
if (entity == null)
{
  return 0;
 }

**Review by Chilly021 — COMMENTED** [review]
`using System.Collections.Generic;
using System.IO;
using System;
using Kesmai.Server.Network;

namespace Kesmai.Server.Items
{
	public partial class WizardHat : Helmet, ITreasure
	{
		/// <inheritdoc />
		public override int LabelNumber => 6000085;

		/// <inheritdoc />
		public override uint BasePrice => 2000;

		/// <inheritdoc />
		public override int Weight => 80;

		/// <inheritdoc />
		public override int ProtectionFromStun => 13;

        public override int MagicProtection => GetMagicalProtectionFromPlayer(Parent);
		
		/// <inheritdoc />
		public override int ProtectionFromFire => 10;
		
		/// <inheritdoc />
		public override int ProtectionFromIce => 10;
		
		/// <inheritdoc />
		public override bool ProvidesNightVision => true;

		/// <summary>
        /// Initializes a new instance of the <see cref="WizardHat"/> class.
        /// </summary>
        public WizardHat() : base(966)
        {
            
        }

        public int GetMagicalProtectionFromPlayer(MobileEntity entity)
        {
            var magicProtection = 0;

            if (entity == null)
            {
                return magicProtection;
            }

            if (entity is PlayerEntity player)
            {
                var skillLevel = entity.GetSkillLevel(Skill.Magic);
                magicProtection = Math.Round(skillLevel * 2);
            }
            else
            {
                magicProtection = 0;
            }

            return magicProtection;
        }

		 /// <inheritdoc />
        public override void GetDescription(List<LocalizationEntry> entries)
        {
            entries.Add(new LocalizationEntry(6200000, 6200387)); /* [You are looking at] [a spectacular azure wizards hat with bright yellow stars.] */

            if (Identified)
                entries.Add(new LocalizationEntry(6250135)); /* The hat emenates power, and shimmers as you touch it. It contains the spell of Night Vision. */
        }

         /// <inheritdoc />
        public override void Serialize(BinaryWriter writer)
        {
            base.Serialize(writer);

            writer.Write((short)1); /* version */
        }

        public override bool CanEquip(MobileEntity entity)
        {

            if (entity is PlayerEntity player)
                 if (player.Profession == Profession.Wizard || player.Profession == Profession.Thaumaturge || player.Profession == Profession.Thief)
                     return true;

            return false;
         }

        /// <inheritdoc />
        public override void Deserialize(BinaryReader reader)
        {
            base.Deserialize(reader);

            var version = reader.ReadInt16();

            switch (version)
            {
                case 1:
                {
                    break;
                }
            }
        }
	}
}
`

**Comment by Wirednuke** [comment]
Ok, using Chilly's input I made another revision. At the very least I am learning a lot. These changes were based on a discussion I had with lux about overriding MagicProtection and feeding it parent. I didn't quite understand what it meant until I started taking the C# classes lol.

**Review by Chilly021 — COMMENTED** [review]
Casing on line 161 public override int MagicProtection => getMagicalProtectionFromPlayer(Parent);

Needs to be: 

GetMagicalProtectionFromPlayer

**Review by Chilly021 — COMMENTED** [review]
Move ln 207 to 193 (move the other code down). Just to match annotation / comments.


        public WizardHat(Serial serial) : base(serial)
        {
        }

**Review by Chilly021 — APPROVED** [review]

**Review by zrikzlok — APPROVED** [review]

**Review by sproatba — APPROVED** [review]

**Pull request merged**